admiral/pkg/clusters/types.go

```diff
@@ -127,11 +130,185 @@ func (g *globalTrafficCache) Put(gtp *v1.GlobalTrafficPolicy) error {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+	// delete all envoyFilters for a given identity+env key
+	delete(r.filterCache, identityEnvKey)
+}
+func (r RoutingPolicyHandler) Added(obj *v1.RoutingPolicy) {
+	if common.ShouldIgnoreResource(obj.ObjectMeta) {
+		log.Infof(LogErrFormat, "success", "routingpolicy", obj.Name, obj.ClusterName, "Ignored the RoutingPolicy because of the annotation")
+		return
+	}
+	dependents, done := getDependents(obj, r)
+	if done {
+		return
+	}
+	r.processroutingPolicy(dependents, obj, admiral.Add)
+	log.Info("finished processing routing policy")
+}
+
+func (r RoutingPolicyHandler) processroutingPolicy(dependents map[string]string, routingPolicy *v1.RoutingPolicy, eventType admiral.EventType ) {
+	for _, remoteController := range r.RemoteRegistry.RemoteControllers {
+
+			for _, dependent := range dependents {
+
+			// Check if the dependent exists in this remoteCluster. If so, we create an envoyFilter with dependent identity as workload selector
+			if _, ok := r.RemoteRegistry.AdmiralCache.IdentityClusterCache.Get(dependent).Copy()[remoteController.ClusterID]; ok {
+
+				filter, err := createOrUpdateEnvoyFilter(remoteController, routingPolicy, eventType, dependent, r.RemoteRegistry.AdmiralCache)
+				if err != nil {
+					log.Errorf(LogErrFormat, admiral.Add, "routingpolicy", routingPolicy.Name, remoteController.ClusterID, err)
+				} else {
+					log.Infof("msg=%s name=%s cluster=%s", "created envoyfilter", filter.Name, remoteController.ClusterID)
+				}
+			}
+		}
+
+	}
+}
+
+func (r RoutingPolicyHandler) Updated(obj *v1.RoutingPolicy) {
+	if common.ShouldIgnoreResource(obj.ObjectMeta) {
+		log.Infof(LogFormat, admiral.Update, "routingpolicy", obj.Name, obj.ClusterName, "Ignored the RoutingPolicy because of the annotation")
+		// We need to process this as a delete event.
+		r.Deleted(obj)
+		return
+	}
+	dependents, missingIdentityLabel := getDependents(obj, r)
+	if missingIdentityLabel {
+		return
+	}
+	r.processroutingPolicy(dependents, obj, admiral.Update)
+	log.Info("updated routing policy")
+}
+
+func getDependents(obj *v1.RoutingPolicy, r RoutingPolicyHandler) (map[string]string, bool) {
+	sourceIdentity := common.GetRoutingPolicyIdentity(obj)
+	if len(sourceIdentity) == 0 {
+		err := errors.New("identity label is missing")
+		log.Warnf(LogErrFormat, "add", "RoutingPolicy", obj.Name, r.ClusterID, err)
+		return nil, true
+	}
+
+	dependents := r.RemoteRegistry.AdmiralCache.IdentityDependencyCache.Get(sourceIdentity).Copy()
+	return dependents, false
+}
+
+func (r RoutingPolicyHandler) Deleted(obj *v1.RoutingPolicy) {
+	log.Info("deleted envoy filter for routing policy - ",obj.Name)
+	dependents, missingIdentityLabel := getDependents(obj, r)
+	if !missingIdentityLabel {
+		r.deleteEnvoyFilters(dependents, obj, admiral.Delete)
+	}
+}
+
+func (r RoutingPolicyHandler) deleteEnvoyFilters(dependents map[string]string, obj *v1.RoutingPolicy, eventType admiral.EventType) {
```

Comment:
@vinay-g is it possible to return the `error` to the caller instead of logging it in this func. I feel the logging should happen at the caller.